src/Lykke.Job.CandlesHistoryWriter/DependencyInjection/JobModule.cs

```diff
@@ -198,19 +198,20 @@ private void RegisterCandles(ContainerBuilder builder)
                 .AutoActivate();
 
             builder.RegisterType<CandlesCacheInitalizationService>()
-                .WithParameter(TypedParameter.From(_settings.HistoryTicksCacheSize))
-                .As<ICandlesCacheInitalizationService>();
+                .WithParameter(TypedParameter.From(_settings.HistoryCache.HistoryTicksCacheSize))
+                .As<ICandlesCacheInitalizationService>()
+                .SingleInstance();
 
             builder.RegisterType<CandlesPersistenceQueueSnapshotRepository>()
                 .As<ICandlesPersistenceQueueSnapshotRepository>()
                 .WithParameter(TypedParameter.From(AzureBlobStorage.Create(_dbSettings.ConnectionString(x => x.SnapshotsConnectionString), TimeSpan.FromMinutes(10))));
 
-            builder.RegisterType<RedisCacheTruncator>()
+            builder.RegisterType<RedisCacheCaretaker>()
                 .As<IStartable>()
                 .SingleInstance()
                 .WithParameter(TypedParameter.From(_marketType))
-                .WithParameter(TypedParameter.From(_settings.CacheCleanupPeriod))
-                .WithParameter(TypedParameter.From(_settings.HistoryTicksCacheSize))
+                .WithParameter(TypedParameter.From(_settings.HistoryCache.CacheCheckupPeriod))
+                .WithParameter(TypedParameter.From(_settings.HistoryCache.HistoryTicksCacheSize))
                 .AutoActivate();
```

Comment:
From the task description:

`Handler should be started only after successfull cache initialization`

Reply:
Being auto activated, the service does not act until the initialization finishes.
Even in future, if the timer ticks during the cache initialization process, it will be cancelled.

Reply:
Just use Startup manager to make this deterministic